File: Year-2019/Mar/2019-03-28-150/lb/150.java
import java.util.Stack;

public class EvalRPN {
    /**
     * 使用栈的一般解
     * @param tokens
     * @return
     */
    public int evalRPN(String[] tokens) {
        Stack<String> stack = new Stack<>();
        int m;
        int n;
        for (String str : tokens){
            if ("+".equals(str)){
                m = Integer.parseInt(stack.pop());
                n = Integer.parseInt(stack.pop());
                stack.push(String.valueOf(m+n));
            }else if("-".equals(str)){
                m = Integer.parseInt(stack.pop());
                n = Integer.parseInt(stack.pop());
                stack.push(String.valueOf(n-m));
            }else if ("*".equals(str)){
                m = Integer.parseInt(stack.pop());
                n = Integer.parseInt(stack.pop());
                stack.push(String.valueOf(n*m));
            }else if ("/".equals(str)){
                m = Integer.parseInt(stack.pop());
                n = Integer.parseInt(stack.pop());
                stack.push(String.valueOf(n/m));
            }else {
                stack.push(str);
            }
        }
        return Integer.parseInt(stack.pop());
    }
}

File: Year-2019/Mar/2019-03-28-150/watercow/150.c
class Solution {
public:
	int evalRPN(vector<string>& tokens) {
		stack<int> num_stack;
		for (auto i : tokens) {
			if (i == "+") {
				int a = num_stack.top();
				num_stack.pop();
				int b = num_stack.top();
				num_stack.pop();
				num_stack.push(a + b);
			}
			else if (i == "-") {
				int a = num_stack.top();
				num_stack.pop();
				int b = num_stack.top();
				num_stack.pop();
				num_stack.push(b - a);
			}
			else if (i == "*") {
				int a = num_stack.top();
				num_stack.pop();
				int b = num_stack.top();
				num_stack.pop();
				num_stack.push(a * b);
			}
			else if (i == "/") {
				int a = num_stack.top();
				num_stack.pop();
				int b = num_stack.top();
				num_stack.pop();
				num_stack.push(b / a);
			}
			else {
				num_stack.push(atoi(i.c_str()));
			}
		}
		return num_stack.top();
	}
};